### DISCUSSION TITLE
Debugger: trace address of reads and writes

### DISCUSSION THREAD

**Discussion by paulb-nl** [mamedev/discussions#35] [Q&A] [answered]
Is there a way to add the address and data of reads and writes to the tracelog? I have read through the documentation but haven't found a way to do this.

Here is an example with a 68K CPU:
move.w  D0, ($2,A1) "Addr $123456, Data: $1234"

**Comment by rb6502** [comment]
That would definitely be useful, but it'd be a big project to update all of the disassemblers.

**Comment by cuavas** [comment, marked as answer]
Can’t you just use a watchpoint to add memory accesses to the trace log?  For example on a Z80, this would log writes to the trace log:
```
wpset 0x0000,0x10000,w,1,{ tracelog "@%04X = %02X",wpaddr,wpdata ; g }
```

**Reply by paulb-nl** [reply]
Thanks, that works. I wasn't aware that it could do that without halting execution. Maybe I read "will halt execution" in the docs and then stopped reading 😄

**Reply by cuavas** [reply]
Well, it actually does halt, but you can immediately start execution again with a `g` command.  Some of the examples [in the documentation](https://docs.mamedev.org/debugger/watchpoint.html#debugger-command-wpset) demonstrate this trick.
